src/lib/pages/schedules.svelte

```diff
@@ -1,37 +1,37 @@
 <script lang="ts">
   import { noop } from 'svelte/internal';
 
+  import { goto } from '$app/navigation';
   import { page } from '$app/stores';
 
   import SchedulesTableRow from '$lib/components/schedule/schedules-table-row.svelte';
   import SchedulesTable from '$lib/components/schedule/schedules-table.svelte';
+  import ApiPagination from '$lib/holocene/api-pagination.svelte';
   import Button from '$lib/holocene/button.svelte';
   import EmptyState from '$lib/holocene/empty-state.svelte';
   import Input from '$lib/holocene/input/input.svelte';
   import Link from '$lib/holocene/link.svelte';
-  import Loading from '$lib/holocene/loading.svelte';
-  import Pagination from '$lib/holocene/pagination.svelte';
   import TableRow from '$lib/holocene/table/table-row.svelte';
   import { translate } from '$lib/i18n/translate';
-  import { fetchAllSchedules } from '$lib/services/schedule-service';
+  import { fetchPaginatedSchedules } from '$lib/services/schedule-service';
   import { coreUserStore } from '$lib/stores/core-user';
+  import type { ScheduleListEntry } from '$lib/types';
+  import type { ErrorCallback } from '$lib/utilities/request-from-api';
   import { routeForScheduleCreate } from '$lib/utilities/route-for';
 
-  import type { ScheduleListEntry } from '$types';
-
   $: namespace = $page.params.namespace;
 
-  let hasSchedules = false;
-
-  $: fetchSchedules = fetchAllSchedules(namespace).then((res) => {
-    const { schedules } = res;
-    hasSchedules = Boolean(schedules?.length);
-    return res;
-  });
+  let error = '';
+  const onError: ErrorCallback = (err) =>
+    (error =
+      err?.body?.message ??
+      `Error fetching schedules: ${err.status}: ${err.statusText}`);
 
   let coreUser = coreUserStore();
   $: createDisabled = $coreUser.namespaceWriteDisabled(namespace);
 
+  $: onFetch = () => fetchPaginatedSchedules(namespace);
+
   let search = '';
   $: filteredSchedules = (schedules: ScheduleListEntry[]) =>
```

Comment:
Just want to clarify that this is expected behavior. When showing filtered schedules the number of items doesn't change (e.g. only two items are shown, but pagination still says 1-3) ⬇️ 
 
<img width="987" alt="Screenshot 2024-01-04 at 12 53 05 PM" src="https://github.com/temporalio/ui/assets/15069288/d7071336-8e44-4bea-8711-3ea7bea50558">

Reply:
We don't get a total from the api. 

For the total when filtering... totally agree it's not ideal but I think for now we'll live with it. There's a whole lot of edge cases if we update the pagination store with a filter.

Reply:
Yeah, I figured it would be a can of worms. I think it's fine for now 👍